src/Symfony/Component/HttpFoundation/File/MimeType/FileBinaryMimeTypeGuesser.php

```diff
@@ -43,7 +43,7 @@ public function __construct($cmd = 'file -b --mime %s 2>/dev/null')
      */
     public static function isSupported()
     {
-        return '\\' !== DIRECTORY_SEPARATOR && function_exists('passthru') && function_exists('escapeshellarg');
+        return '\\' !== DIRECTORY_SEPARATOR && function_exists('passthru') && function_exists('escapeshellarg') && static::hasFileBinary();
```

Comment:
this must be `self::` to access a private API. Otherwise, a child class would break when calling this code (as it would try to call `ChildClass::hasFileBinary`)

Reply:
I've just tried this out. It'd only break when `ChildClass` overrides `hasFileBinary`. But yes, there is no reason for us to use `static::` here, since it's a private method anyway.

Reply:
we could cache the full expression, including the function_exists check (and thus not need any new method btw)

Reply:
Done.